### ISSUE TITLE
KVS Migration Breaks TimestampService Contract

### ISSUE THREAD

**Issue by tpetracca** [palantir/atlasdb#1192]
The timestamp service contract / api states:
```
    /**
     * This will get a fresh timestamp that is guaranteed to be newer than any other timestamp
     * handed out before this method was called.
     */
```

The act of doing a KVS migration with an embedded timestamp service whose timestamp data is co-located in the same KVS and keyspace/database/etc as the Atlas data causes the timestamps to reset back to the logical beginning of time (timestamp 1). This breaks the contract and anyone who might be relying on that contract being true (though it's generally rare to see an embedded TimestampService being used by a client other than Atlas).

I propose we change the logic to the following: At the beginning of the KVS migration, fetch a fresh timestamp from the old timestamp/service and use that timestamp to fast-forward the new TimestampService. Then continue on as usual.

**Comment by clockfort** [comment]
Agreed.
If we want something more extensible to be also used in live KVS migration, suggest instead that the timestamp used is fetched at the end during what would be some atomic swap between KVS A-->B in a live migration.

**Comment by gsheasby** [comment]
Is this still being worked on?

**Comment by rhero** [comment]
Will this be an issue if we go with a Timestamp service impl that stores the timestamp limit locally, or is this an issue with the migration code? I'd rather invest efforts in getting the external Timelock service production ready if that's a tradeoff.

**Comment by tpetracca** [comment]
This problem goes away with external timelock.

**Comment by gsheasby** [comment]
Closed in anticipation of #645 being completed.

**Comment by jboreiko** [comment]
Feels weird to close an active issue in anticipation of another change...

**Comment by gsheasby** [comment]
@rhero for additional context.

We don't intend to spend time specifically on this issue; it will go away with something that's high on our roadmap. If you'd prefer, we could reopen it, put it in the backlog, and add "Fixes #1192" to the external timelock server PR
